diesel_codegen/src/schema_inference/pg.rs

```diff
@@ -38,11 +40,11 @@ table! {
 }
 
 pub fn determine_column_type(cx: &mut ExtCtxt, attr: &ColumnInformation) -> P<ast::Ty> {
-    let tpe = if attr.type_name.starts_with("_") {
-        let subtype = str_to_ident(&capitalize(&attr.type_name[1..]));
+    let tpe = if attr.type_name.ends_with("[]") {
```

Comment:
I considered first to implement this using 2 queries, but decided against it. My concerns with this solution are, that you will need to build the ColumnInformation struct by hand.

If you don't like the current implementation, I can change this back.
Maybe the best solution would be to get this working with two joins.